sdks/python/apache_beam/runners/portability/stager.py

```diff
@@ -119,6 +119,7 @@ def create_job_resources(options,  # type: PipelineOptions
                            temp_dir,  # type: str
                            build_setup_args=None,  # type: Optional[List[str]]
                            populate_requirements_cache=None,  # type: Optional[str]
+                           skip_boot_dependencies=False, # type: Optional[bool]
```

Comment:
Why are "boot" dependencies handled any differently than non-"boot" dependencies?

Reply:
There are other dependencies that doesn't seem to be handled explicitly by the boot sequence, for example pickled main session and java jars for x-lang.

Reply:
It seems the pickled main session should get staged like everything else. The java jars belong to the java environment, and should get staged in the java container image, not here. (But again, maybe this is a Dataflow issue, and will be taken care of when that is cleaned up.)

Reply:
You could consider having one 
```
if not skip_boot_dependencies:
  ...
```
and add a comment/TODO to explain why other dependencies currently cannot be included in this if.